### PR TITLE
[WIP] Run Submission Against Specific Tasks UI

### LINKED ISSUE #540: Missing functionnality: online submission to specific tasks (and not all tasks)
This option is missing on the online submission form while this seems to be a possibility for bot-submissions (see example_submission.py)

### PR DESCRIPTION
**Pull request by Logan-Ruf**
# @ mention of reviewers
@ckcollab 
@gibsonbailey 

# A brief description of the purpose of the changes contained in this PR.
Add an accordion dropdown to let users select what task to run their submission against
image

# Issues this PR resolves
Resolves #540 

# Misc. Comments
Right now any user can run a submission against a selection of tasks. We did limit the visibility of task selection by a field on the competition a user could just send an API Post request manually. On top of that I think it does make sense to enable task selection for all users in-case a user wants to submit to one task for testing, then later submit to all tasks.

# A checklist for hand testing
- [x] Single Task competition does not show accordion
- [x] Single Task competition with fact_sheet works as expected
- [x] Multi Task competition shows accordion and works
- [x] Multi Task competition works with fact_sheet

# Checklist
- [x] Code review by me 
- [x] Hand tested by me 
- [x] I'm proud of my work
- [ ] Code review by reviewer
- [ ] Hand tested by reviewer
- [ ] Ready to merge

### CODE REVIEW

File: docs/example_scripts/example_submission.py [hunk 1]
```diff
@@ -31,8 +31,8 @@
 CODALAB_URL = 'http://localhost/'
 USERNAME = 'admin'
 PASSWORD = 'admin'
-PHASE_ID = None
-TASK_LIST = []
+PHASE_ID = 31
+TASK_LIST = [36, 37]
```
Comment (ckcollab) [inline docs/example_scripts/example_submission.py:35]: these shouldn't come pre-set I don't think? maybe add an assertion somewhere below:

```python
assert PHASE_ID, "Missing PHASE_ID. Please fill in all of the variables at the top of this file."
assert len(TASK_LIST) > 0, "Missing TASK_LIST entries. Please fill in all of the variables at the top of this file."
```

Maybe overkill..

File: src/static/riot/competitions/detail/submission_upload.tag [hunk 2]
```diff
@@ -338,6 +361,16 @@
         self.upload = function () {
             self.display_output = true
 
+            let checkbox_answers = $('#select_tasks_accordion').find('.ui.checkbox').checkbox('is checked')
+            let task_ids_to_run = []
+            for(let i = 0; i < self.selected_tasks.length; i++){
+                console.log("checkbox answer", i, checkbox_answers[i])
```
Comment (ckcollab) [inline src/static/riot/competitions/detail/submission_upload.tag:367]: pull console logs and such ?

**Review by ckcollab — CHANGES_REQUESTED** [review]

**Comment by ckcollab** [comment]
image
may not need to show this if there's only 1 available task?

File: docs/example_scripts/example_submission.py (hunk 1 above)
Comment (Logan-Ruf) [reply docs/example_scripts/example_submission.py:35]: @gibsonbailey already did this on line 47, and an empty task list runs the submission on all of the tasks on a phase.

File: src/static/riot/competitions/detail/submission_upload.tag [hunk 3]
```diff
@@ -5,7 +5,7 @@
         <div class="submission-form">
             <h1>Submission upload</h1>
             <form class="ui form coda-animated {error: errors}" ref="form" enctype="multipart/form-data">
-                <div class="submission-form" if="{ opts.fact_sheet !== null}">
+                <div class="submission-form" ref="fact_sheet_form" if="{ opts.fact_sheet !== null}">
```
Comment (ckcollab) [inline src/static/riot/competitions/detail/submission_upload.tag:8]: nice :)

**Review by ckcollab — APPROVED** [review]

**Pull request merged**